src/features/events/components/EventChangeCampaignDialog.tsx

```diff
@@ -0,0 +1,206 @@
+  const router = useRouter();
+  const { showSnackbar } = useContext(ZUISnackbarContext);
+  const [error, setError] = useState(false);
+
+  const { changeEventCampaign } = useEventMutations(
+    event.organization.id,
+    event.id
+  );
+
+  const fullScreen = useMediaQuery(theme.breakpoints.down('md'));
+
+  const [campaignFilter, setCampaignFilter] = useState('');
+  const [isLoadingCampaign, setIsLoadingCampaign] = useState(0);
+
+  const { data: campaigns } = useCampaigns(event.organization.id);
+  campaigns?.reverse();
+
+  const filteredCampaigns = campaigns
+    ?.filter((campaign) =>
+      campaign.title.toLowerCase().includes(campaignFilter)
+    )
+    .filter((campaign) => campaign.id !== event.campaign?.id);
+
+  const onSearchChange = (value: string) => {
+    setCampaignFilter(value);
+  };
+
+  const handleMove = async (campaign: ZetkinCampaign) => {
+    setIsLoadingCampaign(campaign.id);
+
+    try {
+      changeEventCampaign(campaign.id);
+
+      showSnackbar(
+        'success',
+        messages.eventChangeCampaign.success({
+          campaignTitle: campaign.title,
+        })
+      );
+
+      router.push(
+        `/organize/${campaign.organization.id}/projects/${campaign.id}/events/${event.id}`
+      );
+
+      handleClose();
+    } catch (error) {
+      setIsLoadingCampaign(0);
+      setError(true);
+    }
+  };
+
+  const handleClose = () => {
+    setIsLoadingCampaign(0);
+    setCampaignFilter('');
+    close();
+  };
+
+  return (
+    <Dialog
+      fullScreen={fullScreen}
+      fullWidth
+      maxWidth={'sm'}
+      onClose={() => {
+        close();
+      }}
+      open={isOpen}
+    >
+      <DialogContent
+        sx={{
+          display: 'flex',
+          flexDirection: 'column',
+          height: '85vh',
+        }}
+      >
+        <Box display="flex" justifyContent="space-between">
+          <DialogTitle sx={{ paddingLeft: 2 }} variant="h5">
+            {messages.eventChangeCampaign.dialogTitle()}
+          </DialogTitle>
+
+          <IconButton onClick={handleClose}>
+            <Close
+              color="secondary"
+              sx={{
+                cursor: 'pointer',
+              }}
+            />
+          </IconButton>
+        </Box>
+
+        <Box display="flex" flexDirection="column" rowGap={1}>
+          <TextField
+            fullWidth
+            id="EventMoveDialog-inputField"
+            InputProps={{
+              startAdornment: <Search color="secondary" />,
+            }}
+            onChange={(ev) => onSearchChange(ev.target.value)}
+            value={campaignFilter}
+            variant="outlined"
+          />
+
+          {error && (
+            <Alert severity="error">
+              <Msg id={messageIds.eventChangeCampaign.error} />
+            </Alert>
+          )}
+
+          <Box
+            sx={{
+              overflowY: 'scroll',
+            }}
+          >
+            <List>
+              {Array.isArray(filteredCampaigns) &&
```

Comment:
Is `Array.isArray()` really necessary here? Is it because `filteredCampaigns` might be null? In that case, would `filteredCampaigns?.map()` suffice?